sklearn/kernel_approximation.py

```diff
@@ -782,7 +793,7 @@ def fit(self, X, y=None):
         S = np.maximum(S, 1e-12)
         self.normalization_ = np.dot(U / np.sqrt(S), V)
         self.components_ = basis
-        self.component_indices_ = inds
+        self.component_indices_ = basis_inds
```

Comment:
Why this change?

Reply:
`inds` won't exist if `landmarks` is not `None`. Though, to avoid changing the definition of `self.component_indices_` in the random selection case, we should probably set `inds` in the case of specified landmarks to something like `np.arange(0, n_samples)` and revert this line to `self.component_indices_ = inds`

Reply:
In fact, I don't understand why they even kept `inds` and not `basis_inds` in the original version. It just turns out that they save all sample numbers at all, just mixed in. Why do this I do not fully understand.

Reply:
Fixed it